pl_bolts/datamodules/sklearn_datamodule.py

```diff
@@ -1,280 +1,177 @@
-from pl_bolts.utils.warnings import warn_missing_pkg
+from pl_bolts.datasets import ArrayDataset
 
-if _SKLEARN_AVAILABLE:
-    from sklearn.utils import shuffle as sk_shuffle
-else:  # pragma: no cover
-    warn_missing_pkg("sklearn")
 
-
-@under_review()
-class SklearnDataset(Dataset):
-    """Mapping between numpy (or sklearn) datasets to PyTorch datasets.
-
-    Example:
-        >>> from sklearn.datasets import load_diabetes
-        >>> from pl_bolts.datamodules import SklearnDataset
-        ...
-        >>> X, y = load_diabetes(return_X_y=True)
-        >>> dataset = SklearnDataset(X, y)
-        >>> len(dataset)
-        442
-    """
-
-    def __init__(self, X: np.ndarray, y: np.ndarray, X_transform: Any = None, y_transform: Any = None) -> None:
-        """
-        Args:
-            X: Numpy ndarray
-            y: Numpy ndarray
-            X_transform: Any transform that works with Numpy arrays
-            y_transform: Any transform that works with Numpy arrays
-        """
-        super().__init__()
-        self.X = X
-        self.Y = y
-        self.X_transform = X_transform
-        self.y_transform = y_transform
-
-    def __len__(self) -> int:
-        return len(self.X)
-
-    def __getitem__(self, idx) -> Tuple[np.ndarray, np.ndarray]:
-        x = self.X[idx].astype(np.float32)
-        y = self.Y[idx]
-
-        # Do not convert integer to float for classification data
-        if not ((y.dtype == np.int32) or (y.dtype == np.int64)):
-            y = y.astype(np.float32)
-
-        if self.X_transform:
-            x = self.X_transform(x)
-
-        if self.y_transform:
-            y = self.y_transform(y)
-
-        return x, y
-
-
-@under_review()
-class TensorDataset(Dataset):
-    """Prepare PyTorch tensor dataset for data loaders.
-
-    Example:
-        >>> from pl_bolts.datamodules import TensorDataset
-        ...
-        >>> X = torch.rand(10, 3)
-        >>> y = torch.rand(10)
-        >>> dataset = TensorDataset(X, y)
-        >>> len(dataset)
-        10
-    """
-
-    def __init__(self, X: Tensor, y: Tensor, X_transform: Any = None, y_transform: Any = None) -> None:
-        """
-        Args:
-            X: PyTorch tensor
-            y: PyTorch tensor
-            X_transform: Any transform that works with PyTorch tensors
-            y_transform: Any transform that works with PyTorch tensors
-        """
-        super().__init__()
-        self.X = X
-        self.Y = y
-        self.X_transform = X_transform
-        self.y_transform = y_transform
-
-    def __len__(self) -> int:
-        return len(self.X)
-
-    def __getitem__(self, idx) -> Tuple[Tensor, Tensor]:
-        x = self.X[idx].float()
-        y = self.Y[idx]
-
-        if self.X_transform:
-            x = self.X_transform(x)
-
-        if self.y_transform:
-            y = self.y_transform(y)
-
-        return x, y
-
-
-@under_review()
 class SklearnDataModule(LightningDataModule):
```

Comment:
Btw, one idea how this could again become `sklearn` datamodule (i.e. actually use some indispensable functionality from `sklearn`) might be to try to add a `DataModule`, that would automatically load sklearn dataset (https://scikit-learn.org/stable/datasets.html). This would have to be named differently (something like `SklearnNamedDataModule`) and it's absolutely a new feature, but it's just an idea :rocket:

Reply:
Definitely. I am happy to do this 🚀